### PR TITLE
Update `phpstan/phpstan` from 1.7.4 to 1.8.11

### PR DESCRIPTION
**Pull request by Progi1984**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Bump `phpstan/phpstan` from 1.7.4 to 1.8.10
| Type?             | improvement
| Category?         | CO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | NA
| How to test?      | N/A

### CODE REVIEW

**Comment by Progi1984** [comment]
Ok. So the crash of the CI doesn't work since `1.7.5`.

**Comment by Progi1984** [comment]
Created a bug in the PHPStan project : https://github.com/phpstan/phpstan/issues/7647

File: classes/controller/FrontController.php
Comment (kpodemski) [inline classes/controller/FrontController.php]: Was it reported by PHPStan? What was the reason?

File: classes/Link.php
Comment (atomiix) [inline classes/Link.php]:
```suggestion
            || is_string($category) && ctype_digit($category))
```

File: classes/controller/FrontController.php
Comment (atomiix) [inline classes/controller/FrontController.php]: I'm not sure to understand why you removed it here 🤔

Comment (Progi1984) [reply classes/controller/FrontController.php]: Yes, it was reported by PHPStan.

The reason was that `substr($assign_value, 0, 1) == '/'` was always `true`. It's normal because `assign_array` are a string array with path so PHPStan is right.

Comment (Progi1984) [reply classes/controller/FrontController.php]: Check the comment that I give to @kpodemski : https://github.com/PrestaShop/PrestaShop/pull/29087#discussion_r973119165. (I close this one).

File: classes/Link.php
Comment (Progi1984) [reply classes/Link.php]: (I check it)

Comment (Progi1984) [reply classes/Link.php]: That doesn't work because the PHPDoc is `CategoryCore|array|int` for `$category`

File: classes/controller/FrontController.php
Comment (atomiix) [reply classes/controller/FrontController.php]: Well I guess you should do this then:
```diff
            foreach ($assign_array as $assign_key => $assign_value) {
-               if (substr($assign_value, 0, 1) == '/' || $this->ssl) {
-                   $urls[$assign_key] = $http . Tools::getMediaServer($assign_value) . $assign_value;
+               $urls[$assign_key] = $http . Tools::getMediaServer($assign_value) . $assign_value;
-               } else {
-                   $urls[$assign_key] = $assign_value;
-               }
            }
```

Comment (Progi1984) [reply classes/controller/FrontController.php]: Fixed. Thanks @atomiix

File: classes/Link.php
Comment (atomiix) [reply classes/Link.php]: Then you can't use `ctype_digit` as starting with php 8.1 it only allows `string` as argument

Comment (Progi1984) [reply classes/Link.php]: Oh !!!!!!!!! It's fixed.

**Review by atomiix — DISMISSED** [review]

**Comment by Progi1984** [comment]
@atomiix I bump the version to `1.8.6`

File: composer.lock
Comment (atomiix) [inline composer.lock:12573]: Maybe we should keep 2.2.0?

Comment (Progi1984) [reply composer.lock:12573]: I don't know. Why using an old api version ?

Comment (atomiix) [reply composer.lock:12573]: Because there's still a lot of projects that are not compatible with composer 2.3 and forcing them to use composer 2.3 make us switch between composer version

**Review by FabienPapet — DISMISSED** [review]

File: composer.lock
Comment (Progi1984) [reply composer.lock:12573]: @atomiix Reverted.

**Review by atomiix — DISMISSED** [review]

**Comment by Progi1984** [comment]
@atomiix Sorry for the rebase : but the `1.8.10` is out. Could you review it ?

File: composer.lock
Comment (atomiix) [reply composer.lock:12573]: Same issue again 😅

Comment (Progi1984) [reply composer.lock:12573]: @atomiix Reverted.

**Review by atomiix — DISMISSED** [review]

**Review by zuk3975 — DISMISSED** [review]

**Comment by boubkerbribri** [comment]
@Progi1984 There is some CI workflow that are not running here.
Can we fix that before running automated tests on this PR

**Comment by kpodemski** [comment]
Close & reopen to restart CI.

**Comment by kpodemski** [comment]
@Progi1984 I think you need to rename your PR to avoid skipping the tests.

**Comment by Progi1984** [comment]
@boubkerbribri Rebased

@kpodemski Title renamed

@PrestaShop/prestashop-maintainers & @PrestaShop/committers / @atomiix & @zuk3975 : Your approvals are dismissed. Please review :)

**Comment by Progi1984** [comment]
Automated Tests are executed : https://github.com/Progi1984/ga.tests.ui.pr/actions/runs/3362122163/jobs/5573420171

Errors are fixed in 8.0.x branch :wink:

File: src/PrestaShopBundle/Routing/Converter/LegacyParametersConverter.php
Comment (FabienPapet) [inline src/PrestaShopBundle/Routing/Converter/LegacyParametersConverter.php:85]: missing isset ?

File: src/PrestaShopBundle/Install/Install.php
Comment (FabienPapet) [inline src/PrestaShopBundle/Install/Install.php:877]: where is this condition now ?

Comment (Progi1984) [reply src/PrestaShopBundle/Install/Install.php:877]: Check the l.849.... If `!isset($data['admin_email']`, then go l.871

File: src/PrestaShopBundle/Routing/Converter/LegacyParametersConverter.php
Comment (Progi1984) [reply src/PrestaShopBundle/Routing/Converter/LegacyParametersConverter.php:85]: No. `$legacyParameters['controller']` exists, check l.79-80.

File: composer.lock
Comment (atomiix) [reply composer.lock:12573]: I'm gonna stop fighting for this one as it's already on the 8.0.x branch 🥲

**Review by atomiix — APPROVED** [review]

File: composer.lock
Comment (Progi1984) [reply composer.lock:12573]: Big sorry.

**Review by FabienPapet — APPROVED** [review]

**Review by ghost — APPROVED** [review]

**Review by zuk3975 — APPROVED** [review]

**Comment by sarahdib** [comment]
Hello @Progi1984 

This PR can be merged after if automation test is green. 

Thank you

**Review by jolelievre — APPROVED** [review]

**Comment by jolelievre** [comment]
Thanks @Progi1984

**Pull request merged**
